### PR TITLE
Arknav Protocol Decoder fix

### PR DESCRIPTION
**Pull request by 4lazar**
353425142654345,05*850,000,**PT33**,A,2459.3640,N,12125.2958,E,000.0,224.8,00.8,07:47:26 09-09-05,**1.00,80**

IMEI number,
ID Code (6 Bytes in ASCII), 
Status code (3 Bytes in ASCII), 
Unit number (4 Bytes in ASCII), 
Status (A or V), 
Latitude, N/S indicator,
Longitude, 
E/W Indicator, 
speed, 
Course, 
HDOP, 
GPS time lat/lon value - GPS date lat/lon value, 
Unit Version number, 
battery power level

### CODE REVIEW

**Comment by tananaev** [comment]
Please provide the protocol documentation.

**Comment by tananaev** [comment]
And also add a unit test with a real device sample.

**Comment by 4lazar** [comment]
> Please provide the protocol documentation.

[personal tracker manual V2.pdf](https://github.com/user-attachments/files/17003217/personal.tracker.manual.V2.pdf)

**Comment by 4lazar** [comment]
> d a unit test with a real device sample.
2024-09-14 19:33:15  INFO: [T33904503: arknav < xxx.xxx.xxx.xxx] 355689018xxxxxx,05*827,000,PT33,V,4901.4387,N,02950.7889,E,000.0,000.0,00.0,16:33:14 14-09-24,2.35,16\r

**Comment by tananaev** [comment]
You have to add it as a unit test. Also make sure you're not breaking anything else. It seems like you didn't even run current unit tests. They are failing. Make sure you run `gradle check`.

**Comment by 4lazar** [comment]
I have made an update to the code, there is no errors on check.
New feature was added with battery level and it is displayed in the UI.
image

**Comment by tananaev** [comment]
CI is failing, as you can clearly see, which means that checks are failing. You also haven't added a unit test for your changes.

**Comment by 4lazar** [comment]
Code and unit test was changed.
"gradle check" was executed without errors

File: src/main/java/org/traccar/protocol/ArknavProtocolDecoder.java [hunk 1]
```diff
@@ -76,6 +80,12 @@ protected Object decode(
         position.set(Position.KEY_HDOP, parser.nextDouble(0));
 
         position.setTime(parser.nextDateTime(Parser.DateTimeFormat.HMS_DMY));
+	
+	int battery = 0;
+	if (model = "PT33") {
+		battery = parser.nextInt();
+	}
+	position.set(Position.KEY_BATTERY_LEVEL, battery);
```
Comment (tananaev) [inline src/main/java/org/traccar/protocol/ArknavProtocolDecoder.java:88]: This doesn't make sense. Why would we set it to 0 by default?

File: src/main/java/org/traccar/protocol/ArknavProtocolDecoder.java [hunk 2]
```diff
@@ -47,6 +47,8 @@ public ArknavProtocolDecoder(Protocol protocol) {
             .number("(d+.?d*),")                 // hdop
             .number("(dd):(dd):(dd) ")           // time (hh:mm:ss)
             .number("(dd)-(dd)-(dd),")           // date (dd-mm-yy)
+            .number("d+.?d*,")                   // Unit Version number
+            .number("(.{2})")                      // Battery level
```
Comment (tananaev) [inline src/main/java/org/traccar/protocol/ArknavProtocolDecoder.java:51]:
```suggestion
            .number("(.{2})")                    // Battery level
```

File: src/main/java/org/traccar/protocol/ArknavProtocolDecoder.java [hunk 3]
```diff
@@ -67,6 +69,8 @@ protected Object decode(
         }
         position.setDeviceId(deviceSession.getDeviceId());
 
+	String model = parse.next();
```
Comment (tananaev) [inline src/main/java/org/traccar/protocol/ArknavProtocolDecoder.java:72]: Does this even compile? What is `parse`?

**Comment by tananaev** [comment]
Your code doesn't even compile now. Clearly you didn't run gradle check on it. I'm going to close it because I think the code quality is very low and you didn't even bother to compile it before sending the PR to us.

**Comment by 4lazar** [comment]
Chech it again

**Pull request closed without merging**